File: gam/AS241S3_T15-be-feature-sp3_u3_market_customer 2/src/main/java/pe/edu/vallegrande/project/service/SupermarketCustomerService.java
package pe.edu.vallegrande.project.service;

import pe.edu.vallegrande.project.model.SupermarketCustomer;
import java.util.List;
import java.util.Optional;

public interface SupermarketCustomerService {

    List<SupermarketCustomer> findAll();
    Optional<SupermarketCustomer> findById(Long id);
    SupermarketCustomer save(SupermarketCustomer customer);
    SupermarketCustomer update(SupermarketCustomer customer);
    void delete(SupermarketCustomer customer);
    SupermarketCustomer restore(SupermarketCustomer customer);
}

File: AS241S3_T15-FundoSmart/src/app/core/services/supermarket-customer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { SupermarketCustomer } from '../interfaces/supermarket-customer';

@Injectable({ providedIn: 'root' })
export class SupermarketCustomerService {
  private apiUrl = `${environment.apiUrl}/supermarket-customer`;

  constructor(private http: HttpClient) {}

  getAll(): Observable<SupermarketCustomer[]> {
    return this.http.get<SupermarketCustomer[]>(this.apiUrl);
  }

  getById(id: number): Observable<SupermarketCustomer> {
    return this.http.get<SupermarketCustomer>(`${this.apiUrl}/${id}`);
  }

  save(customer: SupermarketCustomer): Observable<SupermarketCustomer> {
    return this.http.post<SupermarketCustomer>(`${this.apiUrl}/save`, customer);
  }

  update(customer: SupermarketCustomer): Observable<SupermarketCustomer> {
    return this.http.put<SupermarketCustomer>(`${this.apiUrl}/update`, customer);
  }

  delete(id: number): Observable<string> {
    return this.http.delete(`${this.apiUrl}/delete/${id}`, { responseType: 'text' });
  }

  restore(id: number): Observable<string> {
    return this.http.put(`${this.apiUrl}/restore/${id}`, null, { responseType: 'text' });
  }
}
